rfd/0038-database-access-aws-discovery.md

```diff
@@ -0,0 +1,163 @@
+---
+authors: Roman Tkachenko (roman@goteleport.com)
+state: draft
+---
+
+# RFD 38 - Automatic registration and discovery of databases on AWS
+
+## What
+
+Proposes a way for the database service agent to automatically discover and
+register AWS-hosted RDS, Aurora and Redshift databases.
```

Comment:
Outside the scope of this RFD, I've found  [Creating a database account using IAM authentication](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.DBAccounts.html) / Bootstrapping the initial users to be a pain point. 

I could for example use a Terraform provider but then I need to open up ports for the host.  I wonder if we could provide an option to bootstrap an IAM user from the DB host?  This is another [complicated solution](https://blog.alterway.fr/en/bootstrap-your-aurora-schemes-ans-users-at-creation.html) to the problem.

Reply:
Yeah, creating db users kinda disrupts the flow, I agree, though I'm not sure of a good way to simplify this yet cause that requires actually connecting to the database first (unlike, say, Cloud SQL where there's an API for creating users). Redshift supports auto-creating users (which we don't use yet) but RDS doesn't have anything similar as far as I'm aware. We could build a tool into teleport which would connect to RDS endpoint as a one-off with username/password and create an IAM user. Something like `teleport db configure <name-of-rds-database>` akin to `teleport configure`.

Reply:
Yeah, I agree. It's much smoother for Redshift.  I would assume that most people have clusters already defined so it should be less of a step for them.  I don't think we need to add anything yet, we can wait for community feedback.